zerver/actions/users.py

```diff
@@ -383,6 +384,19 @@ def send_update_events_for_anonymous_group_settings(
 
 
 def send_events_for_user_deactivation(user_profile: UserProfile) -> None:
+    subscribed_streams = get_streams_for_user(
+        user_profile,
+        include_public=False,
+        include_subscribed=True,
+    )
```

Comment:
Instead of getting this, technically we could have done some optimisations when we are getting the subscription objects further down the function, but it might make things less readable and more complex for a function that might not be as used as something related to messages.

Reply:
Yeah, this is OK. Deactivation isn't a very common operation that we need to perfectly optimize.